### ISSUE TITLE
Custom CSS Transition doesn’t work

### ISSUE THREAD

**Issue by danielschmid** [barbajs/barba#422]
I’m importing @barba/core and @barba/css as intended but i cant get barba to work with custom CSS classes.

    // tell Barba to use the css module
    barba.use(barbaCss);

    // init Barba
    barba.init({
        transitions:[{
            name: 'custom-transition',
        }]
    });

If my understanding of the documentation is right, barba should add `.custom-transition-appear-active` to the barba wrapper. But barba still uses the basic `.barba-appear-active` class.

Any ideas on this ?

**Comment by danielschmid** [comment]
https://stackoverflow.com/questions/57046532/barba-js-custom-css-transition-doesn-t-work

**Comment by thierrymichel** [comment]
Hi @danielschmid, thanks for reporting.
It should work. I will check and keep you informed!

**Comment by danielschmid** [comment]
I’ve tried a couple of things but with no result. I always get the default classes.

`<main class="main main--page barba-appear-active barba-appear-to" id="main" data-barba="container" data-barba-namespace="about">`

 I’ve also added `cacheIgnore: 'true',` to check if there is any problem with caching an me making changes.

**Comment by danielschmid** [comment]
I’m using the version via the CDN https://unpkg.com/@barba/core and https://unpkg.com/@barba/css

**Comment by danielschmid** [comment]
I’m also using the above code inside a `$(document).ready(function()`

**Comment by thierrymichel** [comment]
Ok. e2e tests are not using the cdn version.
Thanks for the details, I will start investigate from there (but tomorrow in the morning 😊)

**Comment by thierrymichel** [comment]
Hi @danielschmid,

## tl;dr
```
barba.init({
    transitions:[{
        name: 'custom-transition',
        appear() {}, // Fix
    }]
});
```

## detailed explanations

- On `barbaCss` plugin init, a default (`name: 'barba'`) transition is added. It allows to find a transition and use `.barba-hook-xxx` classes.
- On the other hand, to apply an `appear` transition, barba is looking for transition with an `appear` hook.
- In our case `name: 'barba'` is found.

At the moment, I do not have any idea on how to fix this "properly". :(
I would like to avoid `css` logic into the `core` module resolution.

I will update the documentation and keep this issue open.
Thanks again for using barba AND reporting!

**Comment by danielschmid** [comment]
Hi @thierrymichel 
Your fix does the trick. I’ll used it in the meantime.
Thanks

**Comment by thierrymichel** [comment]
#412
